### DISCUSSION TITLE
Is there a way to show formatted files in `ruff format`

### DISCUSSION THREAD

**Discussion by harupy** [astral-sh/ruff#8202] [Q&A] [answered]
Is there a way to show formatted files in `ruff format`?

```sh
> cat a.py
a = ''

# Actual
> ruff format a.py
1 file reformatted

# What I want
> ruff format ??? a.py
reformatted a.py
1 file reformatted
```

**Comment by MichaReiser** [comment]
No, this isn't currently supported. We don't show the file names by default because printing all file names would significantly slow down ruff 😆 

Would you mind to explain your use case? Do you want to use the output in another tool or are is it to verify that Ruff indeed formats all files, or something else?

**Reply by harupy** [reply]
@MichaReiser Thanks for the quick reply!

> because printing all file names would significantly slow down ruff 😆

Makes sense 😆

> Would you mind to explain your use case?

I'm currently using black in https://github.com/mlflow/mlflow. We use [problem-matchers](https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md#problem-matchers) to surface formatting errors on the `Files changed` tab as shown below. I want to do the same thing with ruff.

<img width="762" alt="image" src="https://github.com/astral-sh/ruff/assets/17039389/40f1c2b8-17bc-4a57-b3ed-c6d6ca17f4ce">

**Reply by MichaReiser** [reply, marked as answer]
Oh that's neat. 

`ruff format . --check` shows the names of files that would be reformatted:

```
Would reformat: crates/ruff_cli/resources/test/fixtures/unformatted.py
1 file would be reformatted, 30 files left unchanged
(venv) 
```

But there's no option to show the files of formatted files

**Reply by harupy** [reply]
We could run `git diff --name-only` after `ruff format` to list changed files, but it'd be nice if ruff supports a flag to list them.

**Reply by harupy** [reply]
Looks like `--check` is what I need, thanks!
